elastic/kibana
# [App Search] Curation: Add query management UX & organic documents list

## Summary

The PR expands the single curation view and adds query management UI/UX & the organic documents section:

### Screencaps

![queries](https://user-images.githubusercontent.com/549407/110867780-357ef380-827c-11eb-9123-a727d91d44f4.gif)

Loading states:
<img width="724" alt="" src="https://user-images.githubusercontent.com/549407/110851419-da420680-8265-11eb-903b-5b80027c8b25.png">
<img width="1098" alt="" src="https://user-images.githubusercontent.com/549407/110868230-116fe200-827d-11eb-9425-4961021bc38a.png">

## QA

- Organic results
  - [x] Confirm the Result list looks/behaves as expected from other instances of documents in the app 
- Active query switching
  - [x] Confirm that switching values works / changes the organic results returned
  - [x] Confirm that deleting the currently active query from the manage queries modal causes the active query select to fall back to the first valid query option
- Manage queries modal 
  - [x] Confirm that adding, deleting, or updating a curation query correctly updates the queries select options
  - (Optional) Check/refresh the standalone UI to confirm changes reflect there as well
- Expected error testing:
  - [x] Attempt to add the same query row twice (e.g. `test` followed by `test` again). Confirm you see the following error:
    <img width="838" alt="" src="https://user-images.githubusercontent.com/549407/110852064-bb903f80-8266-11eb-85c6-25348b6b7c4f.png">
  - [x] Attempt to add a query that **already exists** within the a **different** curation set (if in the national parks demo: try `mountains` or `hiking`). Confirm you see the following error:
    <img width="835" alt="" src="https://user-images.githubusercontent.com/549407/110852192-e4b0d000-8266-11eb-8ae7-d20dd6ccd858.png">

### Checklist
